olive/cli/quantize.py

```diff
@@ -0,0 +1,132 @@
+        "pass_flows": [],
+        "cache_dir": "cache",
+        "output_dir": "models",
+        "host": "local_system",
+        "target": "local_system",
+    }
+
+    @staticmethod
+    def register_subcommand(parser: ArgumentParser):
+        sub_parser = parser.add_parser(
+            "quantize",
+            help="Quantize the input model",
+        )
+        sub_parser.add_argument(
+            "-o",
+            "--output_path",
+            type=str,
+            required=True,
+            help="Path to save quantized model weights.",
+        )
+        sub_parser.add_argument(
+            "--algorithms",
+            type=str,
+            nargs="*",
+            required=True,
+            choices=sorted(QuantizeCommand._CONFIG_TEMPLATE["passes"].keys()),
+            help="List of quantization algorithms to run.",
+        )
+
+        # model options
+        QuantizeCommand._add_hf_model_options(sub_parser)
+
+        # dataset options
+        QuantizeCommand._add_dataconfig_options(sub_parser)
+
+        # accelerator options
+        QuantizeCommand._add_accelerator_options(sub_parser)
+
+        sub_parser.set_defaults(func=QuantizeCommand)
+
+    def _get_data_config(self) -> Dict[str, Any]:
+        # NOTE: Can't create DataConfig object here since that prevents validation
+        # which fills in the *required* fields based on the overall run config
+        # but doesn't override if the key already exist. Loading it as dumb json
+        # delays that initialization until the run configuration validation.
+        with open(self.args.data_config_path) as strm:
+            return json.load(strm)
+
+    def _get_run_config(self) -> Dict[str, Any]:
+        data_config = self._get_data_config()
+        to_replace = [
+            (("input_model", "model_path"), self.args.model_name_or_path),
+            (("input_model", "load_kwargs", "trust_remote_code"), self.args.trust_remote_code),
+            (("systems", "local_system", "accelerators", 0, "device"), self.args.device),
+            (("pass_flows"), [[name] for name in self.args.algorithms]),
+            (("output_dir"), self.args.output_path),
+            (("data_configs", [data_config])),
+            (("passes", "gptq", "data_config"), data_config["name"]),
+            (("passes", "quarot", "calibration_data_config"), data_config["name"]),
+        ]
+
+        if self.args.task:
+            to_replace.append((("input_model", "task"), self.args.task))
+
+        if self.args.providers_list:
+            to_replace.append(
+                (("systems", "local_system", "accelerators", 0, "execution_providers"), self.args.providers_list)
+            )
+
+        config = deepcopy(QuantizeCommand._CONFIG_TEMPLATE)
+        for k, v in to_replace:
+            if v is not None:
+                set_nested_dict_value(config, k, v)
+
+        return config
+
+    def run(self):
+        run_config = self._get_run_config()
+        olive_run(run_config)
```

Comment:
Please handle the output like other CLIs and report appropriate msg to the user.